File: dotnet/Scenarios/Exercises/Logic/Repository/Models/BookEntity.cs
namespace EncyclopediaGalactica.Scenarios.Exercises.Logic.Repository.Models;

public class BookEntity
{
    public long Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Authors { get; set; } = string.Empty;
    public long PageStart { get; set; }
    public long PageEnd { get; set; }
    public string Reference { get; set; } = string.Empty;
    public long TopicId { get; set; }
    public TopicEntity? Topic { get; set; }

    public List<ChapterEntity> Chapters { get; set; } = [];
}

File: Rust/eg/src/logic/exercises/repository/entities/book_entity.rs
use sqlx::FromRow;

#[derive(Debug, FromRow)]
pub struct BookEntity {
    pub id: i32,
    #[sqlx(default)]
    pub title: String,
    #[sqlx(default)]
    pub authors: String,
    pub page_start: i32,
    pub page_end: i32,
    pub reference: String,
    pub topic_id: i32,
}
